post/author.go

```diff
@@ -0,0 +1,123 @@
+package post
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	log "github.com/sirupsen/logrus"
+)
+
+func (repository *Repository) GetPostsByAuthor(c *gin.Context) {
+	username := c.Param("username")
```

Comment:
Let's handle bad request error, as API is not having middleware check, it should have proper request validations.